receiver/k8sclusterreceiver/watcher.go

```diff
@@ -95,24 +99,72 @@ func (rw *resourceWatcher) setupInformers(o runtime.Object, informer cache.Share
 }
 
 func (rw *resourceWatcher) onAdd(obj interface{}) {
-	rw.addOrUpdateResource(obj)
+	rw.dataCollector.SyncMetrics(obj)
+
+	// Sync metadata only if there's at least one destination for it to sent.
+	if len(rw.metadataConsumers) == 0 {
+		return
+	}
+
+	newMetadata := rw.dataCollector.SyncMetadata(obj)
+	rw.syncMetadataUpdate(map[string]*collection.KubernetesMetadata{}, newMetadata)
 }
 
 func (rw *resourceWatcher) onDelete(obj interface{}) {
 	rw.dataCollector.RemoveFromMetricsStore(obj)
 }
 
-func (rw *resourceWatcher) onUpdate(_ interface{}, newObj interface{}) {
-	rw.addOrUpdateResource(newObj)
+func (rw *resourceWatcher) onUpdate(oldObj, newObj interface{}) {
+	// Sync metrics from the new object
+	rw.dataCollector.SyncMetrics(newObj)
+
+	// Sync metadata only if there's at least one destination for it to sent.
+	if len(rw.metadataConsumers) == 0 {
+		return
+	}
+
+	oldMetadata := rw.dataCollector.SyncMetadata(oldObj)
+	newMetadata := rw.dataCollector.SyncMetadata(newObj)
+
+	rw.syncMetadataUpdate(oldMetadata, newMetadata)
+}
+
+func (rw *resourceWatcher) setupMetadataExporters(
+	exporters map[configmodels.DataType]map[configmodels.Exporter]component.Exporter,
+	metadataExportersFromConfig []string) error {
+	var out []metadataConsumer
+	metadataExportersSet := utils.StringSliceToMap(metadataExportersFromConfig)
+
+	for pipelineType, exps := range exporters {
+		for cfg, exp := range exps {
+			if !metadataExportersSet[string(cfg.Type())] {
```

Comment:
I think metadata_exporter config values should reference exporters by name instead of type, in case you have multiple and want to only send to one or the other.